integration/spark-common-cluster-test/src/test/scala/org/apache/carbondata/cluster/sdv/generated/CreateTableWithLocalDictionaryTestCase.scala

```diff
@@ -2096,6 +2096,264 @@ class CreateTableWithLocalDictionaryTestCase extends QueryTest with BeforeAndAft
+  {
+    sql("drop table if exists local1")
+    sql(
+      """ | CREATE TABLE local1(id int, name string, city string, age int)
+        | STORED BY 'org.apache.carbondata.format' tblproperties('local_dictionary_enable'='true',
+        | 'local_dictionary_include'='city', 'no_inverted_index'='name')
+      """.stripMargin)
+    val exception = intercept[MalformedCarbonCommandException] {
+      sql(
+        "alter table local1 add columns (alt string,abc int) tblproperties" +
+        "('local_dictionary_include'='abc')")
+    }
+    assert(exception.getMessage
+      .contains(
+        "LOCAL_DICTIONARY_INCLUDE/LOCAL_DICTIONARY_EXCLUDE column: abc is not a String/complex " +
+        "datatype column. LOCAL_DICTIONARY_COLUMN should be no dictionary string/complex datatype" +
+        " column.Please check create table statement."))
+  }
+
+  test("test alter table add column where duplicate columns are present in local dictionary include and exclude")
+  {
+    sql("drop table if exists local1")
+    sql(
+      """
+        | CREATE TABLE local1(id int, name string, city string, age int)
+        | STORED BY 'org.apache.carbondata.format' tblproperties('local_dictionary_enable'='true',
+        | 'local_dictionary_include'='city', 'no_inverted_index'='name')
+      """.stripMargin)
+    val exception = intercept[MalformedCarbonCommandException] {
+      sql(
+        "alter table local1 add columns (alt string,abc string) tblproperties" +
+        "('local_dictionary_include'='abc','local_dictionary_exclude'='alt,abc')")
+    }
+    assert(exception.getMessage
+      .contains(
+        "Column ambiguity as duplicate column(s):abc is present in LOCAL_DICTIONARY_INCLUDE " +
+        "and LOCAL_DICTIONARY_EXCLUDE. Duplicate columns are not allowed."))
+  }
+
+  test("test alter table add column unsupported table property")
+  {
+    sql("drop table if exists local1")
+    sql(
+      """
+        | CREATE TABLE local1(id int, name string, city string, age int)
+        | STORED BY 'org.apache.carbondata.format' tblproperties('local_dictionary_enable'='true',
+        | 'local_dictionary_include'='city', 'no_inverted_index'='name')
+      """.stripMargin)
+    val exception = intercept[MalformedCarbonCommandException] {
+      sql(
+        "alter table local1 add columns (alt string,abc string) tblproperties" +
+        "('local_dictionary_enable'='abc')")
+    }
+    assert(exception.getMessage
+      .contains(
+        "Unsupported Table property in add column: local_dictionary_enable"))
+    val exception1 = intercept[MalformedCarbonCommandException] {
+      sql(
+        "alter table local1 add columns (alt string,abc string) tblproperties" +
+        "('local_dictionary_threshold'='10000')")
+    }
+    assert(exception1.getMessage
+      .contains(
+        "Unsupported Table property in add column: local_dictionary_threshold"))
+  }
+
+  test("test alter table add column when main table is disabled for local dictionary")
```

Comment:
Add one test case in which 2 string columns are added using alter but only one column is specified in local_dic_include. Both should be considered for local dict generation.